sklearn_numba_dpex/common/kernels.py

```diff
@@ -147,66 +147,161 @@ def make_sum_reduction_1d_kernel(size, work_group_size, device, dtype):
     `2 * work_group_size`. This is repeated as many time as needed until only one value
     remains in global memory.
 
-    NB: work_group_size is assumed to be a power of 2.
+    Notes
+    -----
+    `work_group_size` is assumed to be a power of 2.
```

Comment:
What happens if it's not the case? Do you get an error or a invalid result?

Reply:
an invalid result

Reply:
I think we should try to detect it then :)